sdk/servicebus/azure-servicebus/azure/servicebus/_pyamqp/_transport.py

```diff
@@ -177,10 +174,15 @@ def connect(self):
             # are we already connected?
             if self.connected:
                 return
-            self._connect(self.host, self.port, self.connect_timeout)
+            # send in connect timeout of = 1
+            if self.socket_timeout_interval < 1:
+                connect_timeout = 1
+            else:
+                connect_timeout = self.socket_timeout_interval
```

Comment:
I can update this back to a default `connect_timeout` of 1 that is not configurable for now? I updated this to the value that the user passes in and at least 1, b/c we were wondering if connect might be affected by long roundtrips as well (although I can't reproduce it locally). And in that case, would we want to add a separate `connect_timeout` kwarg to the public API?

Reply:
Updating this so that `connect_timeout` is a separate config. Originally, I changed this b/c a customer using Azure Functions was getting a `connection timed out` error that I thought was connected to this kwarg. However, it turned out that they are using websockets, so exposing the `socket_timeout` + passing that in when the websocket connection is being created will fix their problem.